src/main/java/net/glowstone/io/anvil/RegionFile.java

```diff
@@ -200,19 +214,19 @@ public DataInputStream getChunkDataInputStream(int x, int z) throws IOException
         if (version == VERSION_GZIP) {
             byte[] data = new byte[length - 1];
             file.read(data);
-            return new DataInputStream(new GZIPInputStream(new ByteArrayInputStream(data)));
+            return new DataInputStream(new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(data))));
         } else if (version == VERSION_DEFLATE) {
             byte[] data = new byte[length - 1];
             file.read(data);
-            return new DataInputStream(new InflaterInputStream(new ByteArrayInputStream(data)));
+            return new DataInputStream(new BufferedInputStream(new InflaterInputStream(new ByteArrayInputStream(data))));
         }
 
         throw new IOException("Unknown version: " + version);
     }
 
     public DataOutputStream getChunkDataOutputStream(int x, int z) {
         checkBounds(x, z);
-        return new DataOutputStream(new DeflaterOutputStream(new ChunkBuffer(x, z), new Deflater(Deflater.BEST_SPEED)));
+        return new DataOutputStream(new BufferedOutputStream(new DeflaterOutputStream(new ChunkBuffer(x, z))));
```

Comment:
You're writing to a `ChunkBuffer`, which is a `ByteArrayOutputStream` that flushes its output on `close()`. I/O is not the problem. The problem is the amount of deflations, which `BufferedOutputStream` reduces.